### PR TITLE
Change tokensRequested datatype

### PR DESCRIPTION
**Pull request by kangarang**
addresses #2

### CODE REVIEW

File: api/test/app.test.js [hunk 1]
```diff
@@ -242,20 +245,27 @@ describe('POST /api/proposals', () => {
...
       const result = await request(app)
         .post('/api/proposals')
         .send(data);
 
-      // the test db is allowing strings to be added to the db (see: attached screenshot)
-      console.log(Proposal.findAll());
       expect(result.status).toBe(400);
     });
 
+    test('it should return a 200 if `tokensRequested` is a valid number', async () => {
```
Comment (kangarang) [inline api/test/app.test.js]: this is probably redundant

File: api/utils/proposals.js [hunk 2]
```diff
@@ -27,11 +27,14 @@ module.exports = {
     tokensRequested: {
       in: ['body'],
       exists: true,
```
Comment (kangarang) [inline api/utils/proposals.js]: add: custom validation

File: api/utils/proposals.js [hunk 3]
```diff
@@ -27,11 +27,14 @@ module.exports = {
...
-      isDecimal: {
+      trim: true,
```
Comment (kangarang) [inline api/utils/proposals.js]: add custom validator

File: api/test/app.test.js [hunk 4]
```diff
@@ -250,5 +260,20 @@ describe('POST /api/proposals', () => {
         .send(data);
       expect(result.status).toBe(400);
     });
+
+    // TODO: checkSchema(proposalSchema) allows numbers to be added to the db.
```
Comment (akuanti) [inline api/test/app.test.js]: There should be tests here for numbers that are smaller than the base unit, for example, `data.tokensRequested = '100000000000000000'` (`1e17`) should fail.  I see that you're checking on the frontend, but the backend also needs similar checks (you shouldn't be able to POST bad data).

File: api/test/app.test.js [hunk 5]
```diff
@@ -250,5 +260,20 @@ describe('POST /api/proposals', () => {
...
+    test.todo('should not allow test db to add incorrect datatypes');
```
Comment (akuanti) [inline api/test/app.test.js]: I'm unclear what this means, but I don't think there should be separate tests for the tests. If we want to check the data types, we should use `GET /api/proposals` to check that we got the right data out.

File: api/test/app.test.js [hunk 6]
```diff
@@ -233,7 +233,7 @@ describe('POST /api/proposals', () => {
     test.todo('it should return a 4000 if lastName is too long');
 
     // formats
-    test('it should return a 400 if `tokensRequested` is not a number', async () => {
+    test('it should return a 400 if `tokensRequested` is not a valid string (number)', async () => {
```
Comment (akuanti) [inline api/test/app.test.js]: "if `tokensRequested` is a string that cannot be parsed as a number"

File: api/utils/proposals.js [hunk 7]
```diff
@@ -27,9 +29,30 @@ module.exports = {
...
+      isEmpty: false,
+      isLength: {
```
Comment (akuanti) [inline api/utils/proposals.js]: The length check seems redundant given the custom validator.

**Pull request merged**